app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.hidden = hidden
+        self.health = health
+        if health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(animal: Animal) -> None:
+        if isinstance(animal, Carnivore) or animal.hidden:
+            pass
+        else:
+            if animal.health <= 50:
+                Animal.alive.remove(animal)
+            else:
+                animal.health -= 50
```

Comment:
You can simplify your code
```suggestion
    @staticmethod
    def bite(animal: Animal) -> None:
        if isinstance(animal, Carnivore) or animal.hidden:
            return
        animal.health -= 50
        if animal.health <= 50:
                Animal.alive.remove(animal)
```